src/fides/api/util/tcf/tcf_experience_contents.py

```diff
@@ -541,11 +637,35 @@ def get_relevant_systems_for_tcf_attribute(
     # as well as a legal basis of processing of consent or legitimate interests
     starting_privacy_declarations: Query = get_matching_privacy_declarations(db)
```

Comment:
This function is too long, I should revisit at some point.  Its purpose is for consent reporting: we build a "relevant_systems" attribute to be saved alongside a preference. 

Now that we've split things like purpose into purpose consent and purpose legitimate interests, the relevant systems that we save on the consent reporting record need to be further limited by legal basis.

This method is in this file because it shares code with building the experience itself: there's a lot of overlap.